### PR TITLE
[flang][cuda] Add entry point to launch cuda fortran kernel

### PR DESCRIPTION
**Pull request by clementval**

### CODE REVIEW

File: flang/include/flang/Runtime/CUDA/kernel.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+//===----------------------------------------------------------------------===//
+
+#ifndef FORTRAN_RUNTIME_CUDA_KERNEL_H_
+#define FORTRAN_RUNTIME_CUDA_KERNEL_H_
+
+#include "flang/Runtime/entry-names.h"
+#include <cstddef>
+#include <stdint.h>
+
+namespace Fortran::runtime::cuda {
```
Comment (Renaud-K) [inline flang/include/flang/Runtime/CUDA/kernel.h:16]: If they are all `extern "C"` with their own prefix, I don't think they need to be a namespace.

Comment (clementval) [reply flang/include/flang/Runtime/CUDA/kernel.h:16]: Right I can remove that.

Comment (clementval) [reply flang/include/flang/Runtime/CUDA/kernel.h]: Done

**Review by Renaud-K — APPROVED** [review]
Thank you.

**Review by wangzpgi — APPROVED** [review]

**Pull request merged**